nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/AbstractIcebergProcessor.java

```diff
@@ -117,6 +122,15 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         }
     }
 
+    @Override
+    public String getClassloaderIsolationKey(PropertyContext context) {
+        final KerberosUserService kerberosUserService = context.getProperty(KERBEROS_USER_SERVICE).asControllerService(KerberosUserService.class);
+        if (kerberosUserService != null) {
+            return kerberosUserService.getIdentifier();
```

Comment:
I would suggest using the same classloader isolation key as the one in the other hadoop related components: kerberos principal.
```
                final KerberosUser kerberosUser = kerberosUserService.createKerberosUser();
                return kerberosUser.getPrincipal();
```
The controller service identifier also works but it may be too restrictive. It creates separate classloaders for controller service having the same principal but they could share the classloader. The user can create a process group with the kerberos service and the iceberg processor in it and then copy it multiple times. Not the best design because the kerberos service should be extracted in the parent process group in this case but I can imagine it happening.

Reply:
Unfortunately the kerberos principal can't be used here since the controller service is not in enabled state when the isolation key is being retrieved so it will return with null.
As I can see the HDFS processors are using the kerberos principal. It works properly there due the `dynamicallymodifiesclasspath` which is used on the `hadoopConfigPath` property, so the isolation key is going to be evaluated also at processor start. This behaviour is missing here.